airflow/plugins/operators/proto_to_json_operator.py

```diff
@@ -0,0 +1,209 @@
+    fs = get_fs()
+    # get rt files
+    glob = src_path + iso_date + "*"
+    print("Globbing rt bucket {}".format(glob))
+    rt = fs.glob(glob)
+
+    buckets_to_parse = len(rt)
+    print("Realtime buckets to parse: {i}".format(i=buckets_to_parse))
+
+    # organize rt files by itpId_urlNumber
+    rt_files = []
+    for r in rt:
+        rt_files.append(fs.ls(r))
+
+    entity_files = [
+        item for sublist in rt_files for item in sublist if rt_file_substring in item
+    ]
+
+    feed_files = defaultdict(lambda: [])
+
+    for fname in entity_files:
+        itpId, urlNumber = fname.split("/")[-3:-1]
+        feed_files[(itpId, urlNumber)].append(fname)
+
+    # for some reason the fs.glob command takes up a fair amount of memory here,
+    # and does not seem to free it after the function returns, so we manually clear
+    # its caches (at least the ones I could find)
+    fs.dircache.clear()
+
+    # Now our feed files dict has a key of itpId_urlNumber and a list of files to
+    # parse
+    print("found {} feeds to process".format(len(feed_files.keys())))
+    return feed_files
+
+
+def get_google_cloud_filename(filename_prefix, feed, iso_date):
+    itp_id_url_num = "_".join(map(str, feed))
+    prefix = filename_prefix
+    return f"{prefix}_{iso_date}_{itp_id_url_num}.gz"
+
+
+def handle_one_feed(i, feed, files, filename_prefix, iso_date, dst_path):
+    logger = structlog.get_logger().bind(
+        i=i,
+        feed=feed,
+        len_files=len(files),
+        filename_prefix=filename_prefix,
+        iso_date=iso_date,
+        dst_path=dst_path,
+    )
+    logger.info("entering handle_one_feed")
+
+    fs = get_fs()
+
+    if not files:
+        logger.warn("got no files, returning early")
+        return
+
+    google_cloud_file_name = get_google_cloud_filename(filename_prefix, feed, iso_date)
+    logger.info("Creating {} from {} files".format(google_cloud_file_name, len(files)))
+    # fetch and parse RT files from bucket
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        logger.info(f"downloading {len(files)} files to {tmp_dir}")
+        fs.get(files, tmp_dir)
+        all_files = [x for x in Path(tmp_dir).rglob("*") if not x.is_dir()]
+
+        gzip_fname = str(tmp_dir + "/" + "temporary" + ".gz")
+        written = 0
+
+        with gzip.open(gzip_fname, "w") as gzipfile:
+            for feed_fname in all_files:
+                # convert protobuff objects to DataFrames
+                parsed = parse_pb(feed_fname)
+
+                if parsed and "entity" in parsed:
+                    for record in parsed["entity"]:
+                        record["header"] = parsed["header"]
+                        record["calitp_filepath"] = str(parsed["calitp_filepath"])
+                        record["calitp_itp_id"] = int(feed[0])
+                        record["calitp_url_number"] = int(feed[1])
+                        gzipfile.write((json.dumps(record) + "\n").encode("utf-8"))
```

Comment:
question: Did you run into issues with the encoding here at all? (Like, did you add this because you ran into an issue?) Just curious what we know about this.

Reply:
No, the gzip writer just expects bytes